Add Github Action to label issues and PRs

gotmax23 opened the pull request:
This script adds an `sc_approval` label to all PRs that touch files owned by `@ansible/steering-committee` in CODEOWNERS. It also add a `needs_triage` label to new issues and PRs.

This is based on discussion with @samccann and @felixfontein in #ansible-docs earlier today.

This script is relatively simple, but we should be able to easily expand it to add other labels or do other types of triage.

Relates: https://github.com/ansible/ansible-documentation/issues/101

gotmax23 committed:
Add Github Action to label PRs owned by the SC

samccann commented:
+1 .  I can't  really 'review' cuz I barely follow the code but I agree with the strategy.

gotmax23 committed:
labeler: refactor to use ctx obj; add triage label

gotmax23 committed:
labeler: also process new issues

gotmax23 committed:
ci labeler: print event information

gotmax23 commented:
On request from @samccann, I updated this to also add `needs_triage` labels to new issues and PRs. I refactored the code a bit so it could work with both issues and PRs.

gotmax23 commented:
~~Note: someone will need to go into https://github.com/ansible/ansible-documentation/settings/actions and make sure `Require approval for first-time contributors who are new to Github` is set.~~

Disregard. See https://github.com/ansible/ansible-documentation/pull/118#discussion_r1264628855.

felixfontein commented on .github/workflows/labeler.yml:
```diff
@@ -0,0 +1,46 @@
+        run: |
+          echo "${{ toJSON(github.event) }}"
+      - name: Checkout parent repository
+        uses: actions/checkout@v3
```
How about not checking out the repo from the PR, but the current `devel` branch? That way malicious PRs do not result in malicious code being run.

(It does make it harder to update the script, but it also makes the workflow a lot safer. And this can also run automatically for first-time contributors.)

gotmax23 replied:
Definitely! The `pull_request_target` trigger already does this. If the PR targets `devel`, `devel` will be checked out. If the PR targets `stable-2.15`, `stable-2.15` will be checked out, and so on.

felixfontein replied:
Hmm, but then why do you write

> Note: someone will need to go into https://github.com/ansible/ansible-documentation/settings/actions and make sure `Require approval for first-time contributors who are new to Github` is set.

? Doesn't that *not* automatically run this workflow when new contributors create an issue or PR?

gotmax23 replied:
That's the most lenient option, unfortunately. The default is to require approval for *any* new contributors.

felixfontein replied:
Hmm, in gh.com/ansible-collections this is differently: there by default, `pull_request_target` jobs always run. (That's why docs generation works for new contributors, while other tests don't run.)

felixfontein replied:
This seems to be the behavior everywhere according to https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks:

> Note: Workflows triggered by `pull_request_target` events are run in the context of the base branch. Since the base branch is considered trusted, workflows triggered by these events will always run, regardless of approval settings.

gotmax23 replied:
Cool, thanks for verifying that!

gotmax23 commented:
Force pushed to remove an unused TypeVar.

samccann reviewed (approved):
can't comment on code but approach lgtm.

samccann commented:
Explaining the triage label:
1 - I look at open issues with the label - remove the label, add any other labels and add to project board. Might put in enough details to turn it into a `good first issue` as that gets us a lot of help from community.
2. I look at open PRs with the label - remove the label, review/merge or comment to move the PR along.
3. PRs or issues closed w/o removiing the label is fine. I don't look at closed items for that label.
